### PR TITLE
Add EVM tx relay contract

### PR DESCRIPTION
**Pull request by sokyrko**

### CODE REVIEW

**Review by gofmanaa — APPROVED** [review]

File: evm_tx_relay/contracts/Relay.sol [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/utils/cryptography/ECDSA.sol";
+import "@openzeppelin/contracts/utils/cryptography/MessageHashUtils.sol";
+
+contract Relay is Ownable {
+    using MessageHashUtils for bytes32;
+    using ECDSA for bytes32;
+
+    mapping(address => bool) private relayAddresses;
```
Comment (amamoru) [inline evm_tx_relay/contracts/Relay.sol]: rename to validators

File: evm_tx_relay/contracts/Relay.sol [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    uint256 private relayAddressCount = 0;
+    mapping(bytes32 => bool) private relayedTxs;
+
+    constructor() Ownable(msg.sender) {}
+
+    function addRelayAddress(address _relayAddress) public onlyOwner {
```
Comment (amamoru) [inline evm_tx_relay/contracts/Relay.sol]: utilise addValidator

File: evm_tx_relay/contracts/Relay.sol [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    ) public {
+        require(expiration > block.timestamp, "Transaction expired");
+
+        bytes32 txHash = keccak256(abi.encode(target, data, expiration));
+        require(!relayedTxs[txHash], "Transaction already processed");
+
+        bytes32 messageHash = txHash.toEthSignedMessageHash();
+        uint256 validSignatures = 0;
+        for (uint256 i = 0; i < signatures.length; i++) {
+            address signer = messageHash.recover(signatures[i]);
```
Comment (amamoru) [inline evm_tx_relay/contracts/Relay.sol:54]: verify that all signatures are not from same validator more than once

File: evm_tx_relay/test/Relay.ts [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+                await testTarget.getAddress(), data, expiration, [signature1, signature2],
+            );
+
+            const relayRequest2 = relay.relayTransaction(
+                await testTarget.getAddress(), data, expiration, [signature1, signature2],
+            );
+
+            await expect(relayRequest2).to.be.revertedWith("Transaction already processed");
+        });
+    });
```
Comment (amamoru) [inline evm_tx_relay/test/Relay.ts:155]: add test case for several signatures from same validator

File: evm_tx_relay/contracts/Relay.sol (hunk 3 above)
Comment (sokyrko) [reply evm_tx_relay/contracts/Relay.sol:54]: wow, great catch, thanks

**Pull request merged**